plugins/module_utils/dnac.py

```diff
@@ -276,6 +277,23 @@ def check_execution_response_status(self, response):
 
         return self
 
+    def check_string_dictionary(self, task_details_data):
+        """
+        Check whether the input is string dictionary or string.
+
+        Parameters:
+            task_details_data (string) - Input either string dictionary or string.
+
+        Returns:
+            value (dict) - If the input is string dictionary, else returns None.
+        """
+
+        try:
```

Comment:
We are not using self here. Hence we can move this API out of the class.
Also check if the return value is dict using instance.

```
        try:
            value = json.loads(task_details_data)
            if instance(value, dict):
                return value
        except json.JSONDecodeError:
            pass

        return None
```